qiskit_dynamics/pulse/pulse_to_signals.py

```diff
@@ -79,14 +81,60 @@ def __init__(
         self._carriers = carriers or {}
 
     def get_signals(self, schedule: Schedule) -> List[DiscreteSignal]:
-        """
+        r"""Convert a schedule to a corresponding list of DiscreteSignal instances.
+
+        Which channels are converted, and the order they are returned, is controlled by the
+        ``channels`` argument at instantiation. The ``carriers`` instantiation argument sets the
+        analog carrier frequency for each channel, which is fixed for the full duration. For a given
+        channel, the :math:`k^{th}` envelope sample for the corresponding :class:`.DiscreteSignal`
+        is determined according to the following formula:
+
+        .. math::
+            f_k \exp(i(2\pi \Delta\nu(k) k dt + \phi(k) + 2 \pi \phi_a(k))),
+
+        where:
+
+        * :math:`f_k` is the waveform value at the :math:`k^{th}` time step as specified by ``Play``
+          instructions.
+        * :math:`\Delta\nu(k)` is the frequency deviation at time step :math:`k` from the analog
```

Comment:
Just to satisfy my own curiosity, \Delta\nu(k) is always fixed for the duration of a pulse, and only changes between pulses?

Reply:
yeah \Delta\nu(k) can only change with one of the `SetFrequency` or `ShiftFrequency` instructions, and they can't happen during `Play` instruction.